### PR TITLE
#143 unify types and use single hook for application page

### PR DESCRIPTION
**Pull request by nmadruga**

### CODE REVIEW

File: src/utils/types.ts [hunk 1]
```diff
@@ -50,12 +50,19 @@ interface AppStatus {
   outcome: string
 }
 
+interface ElementAndResponse {
+  element: ElementState,
+  response: ResponseFull
+}
```
Comment (CarlosNZ) [inline src/utils/types.ts:56]: This is not being used anymore, is it?

File: src/utils/helpers/validatePage.tsx [hunk 2]
```diff
@@ -41,16 +41,17 @@ interface validatePageProps {
...
 }: validatePageProps): ProgressStatus => {
   let count = 1
 
   // Filter visible elements in the current page
   const visibleQuestions = Object.values(elementsState)
-    .filter(({ section, isVisible }) => isVisible && section === sectionIndex)
+    .filter(({ sectionIndex, isVisible }) => isVisible && currentSectionIndex === sectionIndex)
     .reduce((pageElements: ElementState[], element) => {
-      if (element.elementTypePluginCode === 'pageBreak') count++
+      // TODO: Use the page property from ElementState
```
Comment (CarlosNZ) [inline src/utils/helpers/validatePage.tsx:53]: Did you want to do this for this PR? Probably makes sense if you have time.

**Review by CarlosNZ — COMMENTED** [review]
Great! (I would say that, I was there 😉 )

Just wondering if you want to quickly do the "TO-DO" you left. I think it makes sense to do it as part of this.

File: src/utils/types.ts (hunk 1 above)
Comment (CarlosNZ) [reply src/utils/types.ts:56]: And if it is -- that comma is unnecessary (VSCode should remove it anyway)

**Comment by CarlosNZ** [comment]
Oh, there's one other bug we've inadvertently caused here.

When going back to a page that had previously been filled in, the previous responses are missing. They're being saved to the database okay, but not showing up in the UI.

**Comment by CarlosNZ** [comment]
> Oh, there's one other bug we've inadvertently caused here.
> 
> When going back to a page that had previously been filled in, the previous responses are missing. They're being saved to the database okay, but not showing up in the UI.

Okay, fixed this in d17d9c1.

File: src/utils/types.ts (hunk 1 above)
Comment (CarlosNZ) [reply src/utils/types.ts]: Have removed, along with SectionPages and SectionElements, which are no longer used.

File: src/utils/helpers/validatePage.tsx (hunk 2 above)
Comment (CarlosNZ) [reply src/utils/helpers/validatePage.tsx]: Done this in 994f63f

File: src/utils/types.ts (hunk 1 above)
Comment (nmadruga) [reply src/utils/types.ts]: Yep. Cool! Agreed

**Comment by nmadruga** [comment]
I'll have another look on this PR and check if can be merged.

**Comment by nmadruga** [comment]
There is a problem with going back to the previous page -> The plugins don't get loaded at first. I need to reload the page to make it load correctly. I've created another issue to solve that: #155.

**Comment by CarlosNZ** [comment]
> There is a problem with going back to the previous page -> The plugins don't get loaded at first. I need to reload the page to make it load correctly. I've created another issue to solve that: #155.

It's already solved! 😜
https://github.com/openmsupply/application-manager-web-app/pull/160

**Comment by nmadruga** [comment]
That's nice : )

Can you approve this PR to merge please?

**Review by CarlosNZ — APPROVED** [review]

**Pull request merged**